### PR TITLE
[Backport] Align multiple DEMs with TF (#34)

### PR DESCRIPTION
**Pull request by Jaeyoung-Lim**
**Problem Description**
This is a backport of https://github.com/ethz-asl/grid_map_geo/pull/34 to ROS2

rviz_screenshot_2024_01_18-10_09_17

### CODE REVIEW

**Comment by Ryanf55** [comment]
Want help fixing merge conflicts?

**Comment by Jaeyoung-Lim** [comment]
@Ryanf55 I actually need to fix the part where tf seeped into the grid_map_geo library first

**Comment by Ryanf55** [comment]
> @Ryanf55 I actually need to fix the part where tf seeped into the grid_map_geo library first.
> 
> I would like to keep ros dependencies outside grid_map_geo

Sounds good. If you want to split the package into `grid_map_geo_core` and `grid_map_geo_ros` packages, that could work too.

**Comment by Jaeyoung-Lim** [comment]
@Ryanf55 This should now allow you to download multiple DEMs and Rviz will show them already aligned.

File: include/grid_map_geo/grid_map_geo.hpp
Comment (Ryanf55) [inline include/grid_map_geo/grid_map_geo.hpp:173]:
```suggestion
  geometry_msgs::msg::TransformStamped static_transformStamped_;
```

Nit: Google style guide for class member var

File: launch/load_multiple_tif_launch.xml
Comment (Ryanf55) [inline launch/load_multiple_tif_launch.xml:3]:
```suggestion
```

Unused arg can be removed

File: launch/load_multiple_tif_launch.xml
Comment (Ryanf55) [inline launch/load_multiple_tif_launch.xml:8]: Do you want to instead want me to add a `package.xml` and `CMakeLists.txt` to the `terrain-models` repo so we don't need to symlink across them?

File: src/grid_map_geo.cpp
Comment (Ryanf55) [inline src/grid_map_geo.cpp:61]:
```suggestion
GridMapGeo::GridMapGeo(const std::string& frame_id) { frame_id_ = frame_id; }
```


Pass by const-ref because it's a non-primitive type

File: src/grid_map_geo.cpp
Comment (Ryanf55) [inline src/grid_map_geo.cpp:65]:
```suggestion
bool GridMapGeo::Load(const std::string &map_path, const std::string &color_map_path) {
```

Let's keep them consistent.

File: src/test_tif_loader.cpp
Comment (Ryanf55) [inline src/test_tif_loader.cpp:85]:
```suggestion
    static_transformStamped.transform.translation.x = map_origin.x();
    static_transformStamped.transform.translation.y = map_origin.y();
```

Eigen Vector3d also supports xyz access, which lines up well with the variable it's assigned to.

File: src/test_tif_loader.cpp
Comment (Ryanf55) [inline src/test_tif_loader.cpp:59]: Perhaps we could consider using remappings instead, which don't require a user to parameterize all their topic names.

File: launch/load_multiple_tif_launch.xml
Comment (Jaeyoung-Lim) [reply launch/load_multiple_tif_launch.xml:7]: Maybe, but the data can not be shared by the public, so we should eventually remove references to the terrain-models models in general

File: src/test_tif_loader.cpp
Comment (Jaeyoung-Lim) [reply src/test_tif_loader.cpp:59]: Hmm, but the target remap topic would be provided by the user?

Comment (Ryanf55) [reply src/test_tif_loader.cpp:59]: Yea, or more likely in the launch file.

File: launch/load_multiple_tif_launch.xml
Comment (Ryanf55) [reply launch/load_multiple_tif_launch.xml:7]: Yea, I guess if there is never a way to open source that data, this is good enough.

**Comment by Ryanf55** [comment]
image
Nice work! Just had to zoom out a bit from the default location in the rviz config to see both. A few small things to touch up if you want, otherwise LGTM.

**Comment by Jaeyoung-Lim** [comment]
@Ryanf55 You can actually call sertig and dischma_valley that looks like the following

rviz_screenshot_2024_01_18-03_53_49

File: include/grid_map_geo/grid_map_geo.hpp
Comment (Jaeyoung-Lim) [reply include/grid_map_geo/grid_map_geo.hpp]: Done!

File: src/test_tif_loader.cpp
Comment (Jaeyoung-Lim) [reply src/test_tif_loader.cpp]: Done!

**Comment by Jaeyoung-Lim** [comment]
@Ryanf55 I have tried to address the review comments! Thanks for having a look!

I would probably ignore the style checks and leave it to https://github.com/ethz-asl/grid_map_geo/pull/45

**Review by Ryanf55 — APPROVED** [review]

**Comment by Ryanf55** [comment]
> @Ryanf55 Good to get in?

Looks good!

**Pull request merged**
